micro-rdk-nmea-macros/src/composition.rs

```diff
@@ -141,12 +157,55 @@ impl PgnComposition {
             }
         }
     }
+
+    pub(crate) fn into_fieldset_token_stream(self, input: &DeriveInput) -> TokenStream2 {
+        let name = &input.ident;
+        let parsing_logic = self.parsing_logic;
+        let attribute_getters = self.attribute_getters;
+        let struct_initialization = self.struct_initialization;
+        let proto_conversion_logic = self.proto_conversion_logic;
+        let (impl_generics, src_generics, src_where_clause) = input.generics.split_for_impl();
+        let crate_ident = crate::utils::get_micro_nmea_crate_ident();
+        let mrdk_crate = crate::utils::get_micro_rdk_crate_ident();
+        let error_ident = quote! {#crate_ident::parse_helpers::errors::NmeaParseError};
+        let field_set_ident = quote! {#crate_ident::parse_helpers::parsers::FieldSet};
+
+        quote! {
+            impl #impl_generics #name #src_generics #src_where_clause {
+                #(#attribute_getters)*
+            }
+
+            impl #impl_generics #field_set_ident for #name #src_generics #src_where_clause {
+                fn from_data(cursor: &mut DataCursor) -> Result<Self, #error_ident> {
```

Comment:
It bothers me a bit that `into_token_stream` ends up with ` from_bytes` over a Vec, but `into_fieldset_token_stream` ends up with a `from_data` over a `&mut DataCursor`. Why do they differ that way?

Reply:
I guess they don't need to, I can change `from_bytes` to `from_cursor` and have it take a `DataCursor` argument instead